Fixed kicker float mode, fixed drivetrain encoder inverted, retuned Ramsete for fixed encoder.

ErikCald opened the pull request:
doubleSolenoidFloat must be set to reverse and doubleSolenoidShooter must be set to forward to make float mode work.

This is because of the way float mode is achieved using 2 solenoids.

Testing multiple times and this changed fixed the float mode command.

ErikCald committed:
Fixed kicker float mode

doubleSolenoidFloat must be set to reverse and doubleSolenoidShooter must be set to forward to make float mode work.

This is because of the way float mode is achieved using 2 solenoids.

jogray72 reviewed (approved):
Looks good, we tested and it worked

ErikCald committed:
Fixed drivetrain encoder + retuned auto

I looked at odometry and saw it showing incorrect values. When driving forward the X was not changing.

When driving forward, the right encoder was going positive and the left encoder was going negative.

After fixing the encoder, Ramsete was behaving much more normal. However it meant that the ramsete paths needed to be retuned.

Also added a 4 degree ending angle so that Ramsete's tendency to turn right is handled in code and now Ramsete drives "straight".

CASE 3 auto is broken. The cargo was set to the incorrect pressure. With the correct pressure, it is not mechanically possible to shoot a high goal shot from inside the tarmac.

CASE 4 will go pickup then shoot 2 in high goal. It is identical to case 2 and 3 expect the first shooting command is removed, the rest is the same.
